src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+							{
+								GenerateError(writer, "Support for {0} is not implemented", child.Type.Name);
+							}
+						}
+					}
+				}
+			}
+		}
+
+		private void BuildPropertySetter(IIndentedStringBuilder writer, string fullTargetType, string lineEnding, string property, XamlMemberDefinition valueNode, string targetInstance = null)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			targetInstance = targetInstance ?? "o";
+
+			property = property?.Trim('(', ')');
+
+			// Handle attached properties
+			var isAttachedProperty = property?.Contains(".") ?? false;
+			if (isAttachedProperty)
+			{
+				var separatorIndex = property.IndexOf(".");
+
+				var target = property.Remove(separatorIndex);
+				property = property.Substring(separatorIndex + 1);
+				fullTargetType = FindType(target)?.ToDisplayString();
+
+				if (fullTargetType == null || property == null)
+				{
+					GenerateError(writer, $"Property {property} on {target} could not be found.");
+					return;
+				}
+			}
+
+			var propertyType = FindPropertyType(fullTargetType, property);
+
+			if (propertyType != null)
+			{
+				bool isDependencyProperty = IsDependencyProperty(FindType(fullTargetType), property);
+
+				if (valueNode.Objects.None())
+				{
+					if (isDependencyProperty)
+					{
+						writer.AppendLineInvariant(
+							"new global::Windows.UI.Xaml.Setter({0}.{1}Property, ({2}){3})" + lineEnding,
+							GetGlobalizedTypeName(fullTargetType),
+							property,
+							propertyType,
+							BuildLiteralValue(valueNode, propertyType)
+						);
+					}
+					else
+					{
+						writer.AppendLineInvariant(
+							"new global::Windows.UI.Xaml.Setter<{0}>(\"{1}\", o => {3}.{1} = {2})" + lineEnding,
+							GetGlobalizedTypeName(fullTargetType),
+							property,
+							BuildLiteralValue(valueNode, propertyType),
+							targetInstance
+						);
+					}
+				}
+				else
+				{
+					if (isDependencyProperty)
+					{
+						writer.AppendLineInvariant(
+							"new global::Windows.UI.Xaml.Setter({0}.{1}Property, () => ({2})",
+							GetGlobalizedTypeName(fullTargetType),
+							property,
+							propertyType
+						);
+					}
+					else
+					{
+						writer.AppendLineInvariant(
+							"new global::Windows.UI.Xaml.Setter<{0}>(\"{1}\", o => {2}.{1} = ",
+							GetGlobalizedTypeName(fullTargetType),
+							property,
+								targetInstance
+						);
+					}
+
+					if (HasMarkupExtension(valueNode))
+					{
+						writer.AppendLineInvariant(BuildBindingOption(valueNode, propertyType, prependCastToType: true));
+					}
+					else
+					{
+						BuildChild(writer, valueNode, valueNode.Objects.First());
+					}
+
+					writer.AppendLineInvariant(")" + lineEnding);
+				}
+			}
+			else
+			{
+				GenerateError(writer, "Property for {0} cannot be found on {1}", property, fullTargetType);
+			}
+		}
+
+		private void GenerateError(IIndentedStringBuilder writer, string message)
+		{
+			GenerateError(writer, message.Replace("{", "{{").Replace("}", "}}"), new object[0]);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this call to 'message.Replace' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)